File: src/main/java/com/study/platform/domain/chat/model/ChatMessage.java
package com.study.platform.domain.chat.model;

import com.study.platform.domain.team.model.StudyTeam;
import com.study.platform.domain.user.model.User;
import com.study.platform.global.entity.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Getter
@Entity
@Table(name = "chat_messages")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ChatMessage extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "BINARY(16)")
    public UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id", nullable = false)
    private StudyTeam team;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sender_id", nullable = false)
    private User sender;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Builder
    private ChatMessage(StudyTeam team, User sender, String content) {
        this.team = team;
        this.sender = sender;
        this.content = content;
    }

    public static ChatMessage create(StudyTeam team, User sender, String content) {
        return ChatMessage.builder()
                .team(team)
                .sender(sender)
                .content(content)
                .build();
    }
}

File: domain/src/main/kotlin/com/study/platform/domain/chat/model/ChatMessage.kt
package com.study.platform.domain.chat.model

import com.study.platform.domain.team.model.StudyTeam
import com.study.platform.domain.user.model.User
import com.study.platform.global.entity.BaseTimeEntity
import com.study.platform.global.exception.CustomException
import com.study.platform.global.exception.ErrorCode
import jakarta.persistence.*
import java.util.UUID

@Entity
@Table(name = "chat_messages")
class ChatMessage : BaseTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "BINARY(16)")
    var id: UUID? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id", nullable = false)
    var team: StudyTeam? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sender_id", nullable = false)
    var sender: User? = null

    @Column(nullable = false, columnDefinition = "TEXT")
    var content: String = ""

    companion object {
        private const val MAX_CONTENT_LENGTH = 1000

        fun create(team: StudyTeam, sender: User, content: String): ChatMessage {
            validateContent(content)
            return ChatMessage().also {
                it.team = team
                it.sender = sender
                it.content = content
            }
        }

        private fun validateContent(content: String) {
            if (content.isBlank()) throw CustomException(ErrorCode.INVALID_INPUT)
            if (content.length > MAX_CONTENT_LENGTH) throw CustomException(ErrorCode.INVALID_INPUT)
        }
    }
}
